### PR TITLE
API V2.5.0 Release

### LINKED ISSUE #30: Harmonize POP and intensity
Interesting- it technically shouldn't happen, but I'm not surprised that it does. What I suspect is happening comes from [this line in the ingest script](https://github.com/Pirate-Weather/pirate-weather-code/blob/53b2e6c85dfd9dab074dacb5e475632f02c59997/API/GEFS_Local_Ingest.py#L300), where it finds the number of ensembles where there's more than 1 mm of precipitation. So if a couple members have 0<x<0.1 mm of precipitation, then this could happen. 

The best fix is probably to reduce that POP ingest threshold to something closer to 0.05 mm (I picked 0.1 somewhat arbitrarily), as well as add a check in the response script to intensity to zero if POP == 0, probably in the [hourly calculation ](https://github.com/Pirate-Weather/pirate-weather-code/blob/e6f5c5712548d45ab8d9f6e556aa7dc46ca2e380/API/responseLocal.py#L2484) so that the situation doesn't occur where all the hourly blocks say 0 but there's a daily accumulation.

_Originally posted by @alexander0042 in https://github.com/Pirate-Weather/pirate-weather-code/issues/27#issuecomment-2515552773_

### PR DESCRIPTION
**Pull request by cloneofghosts**
## Describe the change
API release V2.5.0

## Type of change

- [ ] Bugfixes to existing code
- [ ] Breaking change
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation Updates

## Checklist

- This pull request fixes issue: fixes #30 
- [x] Code builds locally. **Your pull request won't be merged unless tests pass**
- [x] Code has been formatted using ruff
- [x] I have signed the CLA agreement

### CODE REVIEW

**Comment by alexander0042** [comment]
I guess I need to sign the CLA too! Didn't even think of that haha

File: Docker/requirements-api.txt [hunk 1]
```diff
@@ -1,21 +1,21 @@
 astral==3.2
-timezonefinder==6.5.5
+timezonefinder==6.5.7
 pytz==2024.2
 zarr==2.18.3
-s3fs==2024.10.0
+s3fs
```
Comment (cloneofghosts) [inline Docker/requirements-api.txt:5]: Any particular reason to remove the version pin here?

File: Docker/requirements-api.txt [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 starlette==0.41.3
 uvicorn==0.32.1
 fastapi==0.115.5
 fastapi-utils==0.8.0
-boto3==1.35.68
+boto3==1.35.73
 scipy==1.14.1
 orjson==3.10.12
-psutil==6.1.0
+psutil==5.9.0
```
Comment (cloneofghosts) [inline Docker/requirements-api.txt:14]: When I setup the file I just used whatever was the latest version was at the time. I'm guessing version 6.1 didn't play nice with the setup?

**Review by cloneofghosts — COMMENTED** [review]
@alexander0042 Just a few minor comments on things in the requirements files.

I've also enabled the ruff linter on this repository and it came back with some errors that could be addressed (see the check section here: https://github.com/Pirate-Weather/pirate-weather-code/actions/runs/12167535966/job/33936325628?pr=31). 

I'm just using the default rules so it's coming back with less errors than if I were to use the HA rules. I also have it setup to not fail at the moment so the formatter can run otherwise the workflow fails without committing any changes.

**Comment by cloneofghosts** [comment]
Meant to commit the lint fixes I did in my branch :laughing:. There are still 27 more lint errors to fix that I haven't addressed here as I'm not as familiar with the code as you are. Have you got it setup yet or you still need help with that?

**Comment by cloneofghosts** [comment]
@alexander0042 Just an FYI that I did some changes with how the repo operates:

1. I created a `dev` branch based on the `main` branch which corresponds to the dev endpoint.
2. I merged this PR into `dev`
3. Changed the default branch to `dev` so the dependencies will target that branch instead.

So basically instead of creating a separate branch for each new update the `dev` branch will be used and once stable a PR will be created against `main` to merge it into the production endpoint.

**Comment by alexander0042** [comment]
This makes a ton of sense- dev is what we use for everything else, so that keeps the wording clear. 

Re: the generic types, I'd removed them, but putting them back in before release. Just going through issues and seeing what else needs to be in before I push it live

**Pull request merged**